testing/scenario_app/assemble_apk.sh

```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
 
+set -e
+
+pushd "${BASH_SOURCE%/*}/../../.."
+  ./flutter/tools/gn --unopt
+  ninja -C out/host_debug_unopt sky_engine sky_services
```

Comment:
@blasten  - I believe this is a better fix than changing the pubspec, since we have other modes that want to run jit and need debug mode built.  The expectation is that everyone runs at least this for host_debug_unopt (it's a pretty fast build step, just involves copying/repackaging dart files).

Reply:
Is there any benefit in doing this step in the LUCI recipe before `assemble_apk.sh` is invoked?

Reply:
At some point that's how it worked I think.  The danger is that someone "fixes" the recipe without realizing its breaking this script - this also lets the script run locally.

If this was doing actual C compilation it could be a problem too, because you have to make sure goma is available/running and LUCI likes to handle that special, but it should be fine here.  These tasks just copy dart files around.